### PR TITLE
Core/Map: implemented additional table to override default zone light id

### PR DESCRIPTION
**Pull request by sanctum32**
**Changes proposed:**

-  implemented default zone light overriding by database
-  
-  

**Target branch(es):** 3.3.5

- [x] 3.3.5

**Issues addressed:** Closes #  (insert issue tracker number)
#24250 

**Tests performed:** (Does it build, tested in-game, etc.)
builds, tested in-game

**Known issues and TODO list:** (add/remove lines as needed)

- None

### CODE REVIEW

**Comment by sanctum32** [comment]
To perform tests
```
DELETE FROM `zone_default_light_override` WHERE `zoneId` IN (4395, 4812, 4500);
INSERT INTO `zone_default_light_override` (`zoneId`, `defaultLight`, `comment`) VALUES 
(4395, 1802, 'Dalaran'),
(4812, 2488, 'Icecrown Citadel - The Frozen Throne'),
(4500, 1773, 'Eye of the Eternity');
```

File: sql/updates/world/3.3.5/9999_99_99_99_world.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+DROP TABLE IF EXISTS `zone_default_light_override`;
+CREATE TABLE `zone_default_light_override` (
+	`zoneId` INT(10) UNSIGNED NOT NULL,
+	`defaultLight` INT(10) UNSIGNED NOT NULL,
+	`comment` VARCHAR(255) NULL DEFAULT NULL COLLATE 'utf8_general_ci',
+	PRIMARY KEY (`zoneId`)
+)COLLATE='utf8_general_ci'ENGINE=InnoDB;
```
Comment (illfated) [inline sql/updates/world/3.3.5/9999_99_99_99_world.sql]: Single space or NewLine after the closing parenthesis ) .

Comment (sanctum32) [reply sql/updates/world/3.3.5/9999_99_99_99_world.sql]: sure

**Review by ghost — APPROVED** [review]

**Comment by Shauren** [comment]
Thank you for the pull request but this is not the correct way to handle it as one zone can have multiple active lights at the same time and only one of the is being overriden

**Comment by sanctum32** [comment]
one word: pointless...

**Pull request closed without merging**